packages/core/schematics/migrations/renderer-to-renderer2/google3/rendererToRenderer2Rule.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Replacement, RuleFailure, Rules} from 'tslint';
+import * as ts from 'typescript';
+
+import {HelperFunction, getHelper} from '../helpers';
+import {migrateExpression, replaceImport} from '../migration';
+import {findRendererImport, findRendererReferences} from '../util';
+
+export class Rule extends Rules.TypedRule {
+  applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): RuleFailure[] {
+    const typeChecker = program.getTypeChecker();
+    const printer = ts.createPrinter();
+    const failures: RuleFailure[] = [];
+    const rendererImport = findRendererImport(sourceFile);
+
+    // If there are no imports for the `Renderer`, we can exit early.
+    if (!rendererImport) {
+      return failures;
+    }
+
+    const {typedNodes, methodCalls} = findRendererReferences(sourceFile, typeChecker);
+    const helpersToAdd = new Set<HelperFunction>();
+
+    failures.push(this._getNamedImportsFailure(rendererImport, sourceFile, printer));
+    typedNodes.forEach(node => failures.push(this._getTypedNodeFailure(node, sourceFile)));
+
+    methodCalls.forEach(call => {
+      const {failure, requiredHelpers} = this._getMethodCallFailure(call, sourceFile, printer);
+
+      failures.push(failure);
+
+      if (requiredHelpers) {
+        requiredHelpers.forEach(helperName => helpersToAdd.add(helperName));
+      }
+    });
+
+    // Some of the methods can't be mapped directly to `Renderer2` and need extra logic around them.
+    // The safest way to do so is to declare helper functions similar to the ones emitted by TS
+    // which encapsulate the extra "glue" logic. We should only emit these functions once per
+    // file and only if they're needed.
+    if (helpersToAdd.size) {
+      failures.push(this._getHelpersFailure(helpersToAdd, sourceFile, printer));
+    }
+
+    return failures;
+  }
+
+  /** Gets a failure for an import of the Renderer. */
+  private _getNamedImportsFailure(
+      node: ts.NamedImports, sourceFile: ts.SourceFile, printer: ts.Printer): RuleFailure {
+    const replacementText = printer.printNode(
+        ts.EmitHint.Unspecified, replaceImport(node, 'Renderer', 'Renderer2'), sourceFile);
+
+    return new RuleFailure(
+        sourceFile, node.getStart(), node.getEnd(), 'Imports of Renderer are not allowed.',
```

Comment:
Can we add some info on what to do? like:

```suggestion
        sourceFile, node.getStart(), node.getEnd(), 'Imports of deprecated Renderer are not allowed. Please use Renderer2 instead. ',
```